src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,146 @@
+    private static final int INVALID_AGE = 15;
+
+    private RegistrationService registrationValidator;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        registrationValidator = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    public void register_existingUser_notOk() {
+        User existingUser = createUser(EXISTING_USER_LOGIN, VALID_PASSWORD, VALID_AGE);
+        storageDao.add(existingUser);
+        assertThrows(RegistrationException.class, ()
+                -> registrationValidator.register(existingUser));
+    }
+
+    @Test
+    public void register_newUser_ok() throws RegistrationException {
+        User newUser = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        registrationValidator.register(newUser);
+        assertEquals(newUser, storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_validLogin_ok() throws RegistrationException {
+        User newUser = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        registrationValidator.register(newUser);
+        assertEquals(newUser, storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_invalidLogin_notOk() {
+        User newUser = createUser(SHORT_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(SHORT_LOGIN));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(SHORT_LOGIN));
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        User newUser = createUser(null, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(null));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(null));
+    }
+
+    @Test
+    public void register_emptyLogin_notOk() {
+        User newUser = createUser(EMPTY_STRING, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(EMPTY_STRING));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(EMPTY_STRING));
+    }
+
+    @Test
+    public void register_validPassword_ok() throws RegistrationException {
+        User newUser = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        registrationValidator.register(newUser);
+        assertEquals(newUser, storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_invalidPassword_notOk() {
+        User newUser = createUser(VALID_LOGIN, SHORT_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(VALID_LOGIN));
```

Comment:
Similar to the login validation, this test case checks for an invalid password but does not specify what makes the password invalid. Ensure that the service contains a clear rule for password validity (such as minimum length) and that this test reflects that rule.